### ISSUE TITLE
How do I implement this featurer?

### ISSUE THREAD

**Issue by xianghx** [joaotavora/yasnippet#818]
I want this feature， see:  https://github.com/SirVer/ultisnips

When I add arguments to the constructor, they automatically get assigned to instance variables. 

Stupid method:  b1/b2/b3， what should I do?
```elisp
# -*- mode: snippet -*-
# name: builder1
# key: b1
# type: snippet
# --
struct ${1:Name}Builder {
    $2: $3,
}

impl $1Builder {
    fn new() -> $1Builder {
        $1Builder { $2: $20, }
    }

    fn $2(&mut self, $2: $3) -> &mut $1Builder {
        self.$2 = $2;
        self
    }

    fn finalize(&self) -> $1 {
        $1 { $2: self.$2 }
    }
}
```

**Comment by npostavs** [comment]
Does https://github.com/joaotavora/yasnippet/issues/766#issuecomment-273356496 help?

**Comment by xianghx** [comment]
Thanks. I write a builder snippet, not perfect. and not work with rust-mode. So I used in the emacs-lisp-mode, then copy it. Hope  someone to improve it.  
```elisp
# -*- mode: snippet -*-
# name: builder
# key: builder
# --
struct ${1:Name}Builder {
${2:age: i32}
}

impl $1Builder {
    fn new() -> $1Builder {
        $1Builder {
${2:$(mapconcat
(lambda (fkv)
(let* ((flist (split-string fkv " *:" t " *"))
(fv (nth 0 flist)))
(concat fv ": "  ",")))
 (split-string yas-text " *," t " *") "\n")}
      }
    }

${2:$(mapconcat
(lambda (fkv)
(let* ((flist (split-string fkv " *:" t " *"))
(fv (nth 0 flist)))
(concat
  "fn " fv "(&mut self, " fkv ") -> &mut " "XXXBuilder {" "\n"
        "self." fv " = " fv";" "\n" 
        "self" "\n"
    "}" "\n")))
 (split-string yas-text " *," t " *") "\n")}

fn finalize(&self) -> $1 {
        $1 { 
${2:$(mapconcat
(lambda (fkv)
(let* ((flist (split-string fkv " *:" t " *"))
(fv (nth 0 flist)))
(concat fv ": self." fv ",")))
 (split-string yas-text " *," t " *") "\n")}
    }
  }
}
```

**Comment by npostavs** [comment]
> I write a builder snippet, not perfect. and not work with rust-mode.

Ah, that was a bug, thanks for reporting. I pushed a fix just now (#820).

**Comment by xianghx** [comment]
Thanks, it can work with rust-mode.
